<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainWindow</class>
 <widget class="QMainWindow" name="mainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>921</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1200</width>
    <height>921</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Лабораторная работа №9</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QLabel" name="label">
        <property name="minimumSize">
         <size>
          <width>320</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>320</width>
          <height>60</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Отсечение многоугольников с помощью &lt;br/&gt;алгоритма &lt;span style=&quot; font-size:12pt; background-color:transparent;&quot;&gt;Сазерленда-Ходжмена&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="minimumSize">
         <size>
          <width>320</width>
          <height>250</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>320</width>
          <height>330</height>
         </size>
        </property>
        <property name="title">
         <string>Добавить регулярный отсекатель</string>
        </property>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>100</y>
           <width>291</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Координаты новой точки</string>
         </property>
        </widget>
        <widget class="QTextEdit" name="clipper_point_x">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>150</y>
           <width>141</width>
           <height>31</height>
          </rect>
         </property>
        </widget>
        <widget class="QTextEdit" name="clipper_point_y">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>150</y>
           <width>141</width>
           <height>31</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>120</y>
           <width>141</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>X</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>120</y>
           <width>141</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QPushButton" name="addClipperPointBtn">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>200</y>
           <width>300</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>Добавить точку</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_16">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>291</width>
           <height>61</height>
          </rect>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;(Также ввод точек отсекателя &lt;br/&gt;производится с помощью нажатия &lt;br/&gt;правой кнопки мыши)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
        <widget class="QPushButton" name="closeClipperBtn">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>240</y>
           <width>300</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>Замкнуть отсекатель</string>
         </property>
        </widget>
        <widget class="QPushButton" name="deleteClipperBtn">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>280</y>
           <width>300</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>Удалить отсекатель</string>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>300</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>320</height>
         </size>
        </property>
        <property name="title">
         <string>Добавить произвольный многоугольник</string>
        </property>
        <widget class="QLabel" name="label_10">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>120</y>
           <width>141</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>X</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QLabel" name="label_11">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>120</y>
           <width>141</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Y</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QTextEdit" name="shape_point_x">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>150</y>
           <width>141</width>
           <height>31</height>
          </rect>
         </property>
        </widget>
        <widget class="QTextEdit" name="shape_point_y">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>150</y>
           <width>141</width>
           <height>31</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_17">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>291</width>
           <height>61</height>
          </rect>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;(Также ввод точек многоугольника&lt;br/&gt;производится с помощью нажатия &lt;br/&gt;левой кнопки мыши)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
        <widget class="QPushButton" name="addShapePointBtn">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>200</y>
           <width>300</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>Добавить точку</string>
         </property>
        </widget>
        <widget class="QPushButton" name="closeShapeBtn">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>240</y>
           <width>300</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>Замкнуть многоугольник</string>
         </property>
        </widget>
        <widget class="QPushButton" name="deleteShapeBtn">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>280</y>
           <width>300</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>Удалить все фигуры</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>100</y>
           <width>291</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Координаты новой точки</string>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>150</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>320</width>
          <height>150</height>
         </size>
        </property>
        <property name="title">
         <string/>
        </property>
        <widget class="QPushButton" name="cutBtn">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>70</y>
           <width>300</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>Произвести отсечение</string>
         </property>
        </widget>
        <widget class="QPushButton" name="clearBtn">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>110</y>
           <width>300</width>
           <height>31</height>
          </rect>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>31</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>300</width>
           <height>31</height>
          </size>
         </property>
         <property name="text">
          <string>Очистить плоскость</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>261</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Время выполнения</string>
         </property>
        </widget>
        <widget class="QLabel" name="timeLabel">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>40</y>
           <width>211</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>0 мс</string>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true">background:white</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <action name="action">
   <property name="text">
    <string>Загрузить данные</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
